src/controller/python/chip-device-ctrl.py

```diff
@@ -388,10 +391,56 @@ def do_blescan(self, line):
 
         return
 
+    def ConnectFromSetupPayload(self, setupPayload, nodeid):
+        # TODO(cecille): Get this from the C++ code?
+        softap = 1 << 0
+        ble = 1 << 1
+        onnetwork = 1 << 2
+        # Devices may be uncommissioned, or may already be on the network. Need to check both ways.
+        # TODO(cecille): implement soft-ap connection.
+
+        if setupPayload.rendezvousInformationFlags & onnetwork:
+            print("Attempting to find device on Network")
+            self.devCtrl.DiscoverCommissioningLongDiscriminator(setupPayload.discriminator)
+            print("Waiting for device responses...")
+            strlen = 100;
+            addrStrStorage = ctypes.create_string_buffer(strlen)
+            count = 0
+            # If this device is on the network and we're looking specifically for 1 device,
+            # expect a quick response.
+            maxWaitTime = 1
+            ok = False
+            while count < maxWaitTime:
+                ok = self.devCtrl.GetIPForDiscoveredDevice(0, addrStrStorage, strlen)
+                if ok:
+                    break
+                time.sleep(0.2)
+                count = count + 0.2
+            if ok:
+                addrStr = addrStrStorage.value.decode('utf-8')
+                print("Connecting to device at " + addrStr)
```

Comment:
What if there are several devices colliding in discriminator? This will always choose the first one responding. This  is OK for now, but it is a caveat that could be documented.

Reply:
Added a comment.